coala_quickstart/generation/Project.py

```diff
@@ -25,3 +26,66 @@ def get_project_dir(printer):
             printer.print("Please enter a valid directory", color="blue")
 
     return project_dir
+
+
+def get_popular_languages(file_paths):
+    """
+    Identifies the most popular languages in the user's project directory
+    from the files matched from the given glob expression.
+
+    :param file_paths:
+        A list of absolute file paths in the user's project directory.
+    :return:
+        A list of tuples containing a language name as the first value
+        and percentage usage in the project as the second value.
+    """
+    results = {}
+    for ext in exts:
+        results[exts[ext]] = 0
+    results["Unknown"] = 0
+
+    for file_path in file_paths:
+        name, ext = os.path.splitext(file_path)
+        if ext in exts:
+            results[exts[ext]] += 1
+        else:
+            results["Unknown"] += 1
+    if len(file_paths):
+        for ext in results:
+            results[ext] = (100.0 * results[ext]) / len(file_paths)
+
+    languages = sorted(
+        results.items(),
+        key=operator.itemgetter(1),
+        reverse=True)
+    used_languages = []
+    append_list = []
+    for lang, percent in languages:
+        if lang == "Unknown":
+            append_list += [(lang, int(percent))]
+            continue
+        if percent == 0:
+            break
+        else:
+            used_languages += [(lang, int(percent))]
```

Comment:
> you can use list.append to append to single items

I have no idea why I'm creating an array of one element. Will do.

> currently your threshold is < 1% , as you convert your float to an int and check then for zero^^

But that's before the int type cast - the type cast happens only in list insertion.

> Yeah that's what I've meant, but maybe you should let the printing function take care of this. Maybe you can use rounding then which is maybe more appropriate 

:+1: